grammar.js

```diff
@@ -174,7 +174,11 @@ module.exports = grammar({
             number_base(2),
             number_base(8),
             number_base(10),
-            number_base(16)))),
+            number_base(16),
+            seq(
+              optional(/[+-]/),
+              /(inf|nan)\.[0ft]/
+            )))),
```

Comment:
This implementation is not compatible with Racket. See [Reading Numbers](https://docs.racket-lang.org/reference/reader.html#%28part._parse-number%29).
For example, `inf.0` is a symbol. But `+Inf.F`, `+Inf.FI` are numbers.

The current definition in `number_base` should work well.

Another reason why you should not put inf and nan definition here is that it's not maintainable.
I'd like to keep number things in `number_base` and exactly reflect the grammar description in the official docs.
So it's easier to read, understand, and change once they update.

Reply:
My bad, the sign is not optional, and I didn't even think that inf and nan can be imaginary numbers. Reverting this change in the next commit.